src/linter/block.go

```diff
@@ -1193,7 +1198,19 @@ func (b *BlockWalker) enterClosure(fun *ir.ClosureExpr, haveThis bool, thisType
 		delete(b.unusedVars, v.Name)
 	}
 
-	params, _ := b.r.parseFuncArgs(fun.Params, phpDocParamTypes, sc)
+	// find the types for the arguments of the function that contains this closure
+	var funcArgs []meta.TypesMap
+	for _, arg := range b.ctx.callerFunctionArgs {
+		tp := solver.ExprType(parentScope, b.r.ctx.st, arg.(*ir.Argument).Expr)
```

Comment:
You cannot call ExprType() during the indexing phase

Reply:
Basically this invalidates the whole implementation. You cannot use solved types during this stage so you can't check that the type of the arguments are only arrays and you don't need to as well.